lib/system/sysstr.nim

```diff
@@ -334,7 +334,8 @@ proc setLengthSeqV2(s: PGenericSeq, typ: PNimType, newLen: int): PGenericSeq {.
       # presence of user defined destructors, the user will expect the cell to be
       # "destroyed" thus creating the same problem. We can destroy the cell in the
       # finalizer of the sequence, but this makes destruction non-deterministic.
-      zeroMem(dataPointer(result, elemAlign, elemSize, newLen), (result.len-%newLen) *% elemSize)
+      if typ.base.kind notin tyInt..tyUInt64: # optimization for trivial types
+        zeroMem(dataPointer(result, elemAlign, elemSize, newLen), (result.len-%newLen) *% elemSize)
```

Comment:
does it zero on growing back though? ie.. 

```nim
var v = newSeq[int](100)
v[99]= 444
v.setLen(5)
v.setLen(100)
echo v[99]
```

Reply:
also, this needs a bit more work - ie `tuple[int, int]` is also a trivial type, as is `object = a, b: int`..

What I've wanted from nim is a type trait similar to `supportsCopyMem` that correctly identifies all trivial types and can be used in implementations like this to select "better" implementations.

Reply:
> does it zero on growing back though?
Yes

Reply:
> What I've wanted from nim is a type trait similar to supportsCopyMem that correctly identifies all trivial types and can be used in implementations like this to select "better" implementations.

`supportsCopyMem` should suffix to identify trivial types (`containsGarbageCollectedRef(t) or hasDestructor(t)`. Probably I can add a new parameter called `isTrivialTypes` so the flag can be passed in the backends

Reply:
> supportsCopyMem

I can imagine there being subtle differences here, or at least in c++ there are (between `is_trivial` and `is_trivially_copyable` .. if there exists a `=copy` for example, what should happen here?

> this PR makes changes 

yes, this is my concern: the PR removes some zeroing but doesn't add any new zeroing - it implies that either things are being zeroed twice before this pr, or that after the pr, things are not being zeroed. there's _at least_ one more bug here ;)

Reply:
The `supportsCopyMem` trait is appropriate for this case. If there exist a `=copy` it does not support `copyMem`. I think you might miss the connection to `system.copyMem` (aka `memcpy`) here.